src/Workspaces/Core/Portable/Workspace/Solution/DocumentInfo.cs

```diff
@@ -154,7 +154,7 @@ internal sealed class DocumentAttributes(
             SourceCodeKind sourceCodeKind,
             string? filePath,
             bool isGenerated,
-            bool designTimeOnly) : IChecksummedObject, IObjectWritable
+            bool designTimeOnly) : IObjectWritable
```

Comment:
thse types implmeented this interface, just to pass to a helper which would check for the interface, then call back into .Checksum.

Remove this and just updated those places to access the checksum directly.